scripts/check_urls.py

```diff
@@ -20,107 +21,172 @@
-    :return: a list of all file matching the given extension
-  """
-  path = path[:-1] if path[-1] == "/" else path
-  files = glob.glob(path + ext, recursive=True)
-  return readFile(files)
+def scan_directory(path, exts=['*.md', '*.MD']):
+    """
+      Scan the chosen directory, and the sub-directories and returns the execution of readFile from the found collection of files
+      :param path: directory to scan
+      :param ext: file extension to retrieve
+      :return: a list of all file matching the given extension
+    """
+    path = path[:-1] if path[-1] == "/" else path
+    files = []
+    for ext in exts:
+        files.extend(list(Path(path).rglob(ext)))
+    return readFile(files)
+
+
+def extract_uris(text):
+    # Anything that isn't a square closing bracket
+    name_regex = "[^]]+"
+    # http:// or https:// followed by anything but a closing paren
+    url_regex = "http[s]?://[^)]+"
+    markup_regex = '\[({0})]\(\s*({1})\s*\)'.format(name_regex, url_regex)
+    result = re.findall(markup_regex, text)
+    return list(filter(lambda r: not r.lower().startswith("ioit://"), map(lambda r: r[1], result)))
 
 
 def readFile(files):
-  """
-  Reads the collection of files passed as parameter and returns the set of uris found inside all the files
-  :param files: an iterable of file paths
-  :return: a set contain all uri found
-  """
-  uris = set()
-  for path in files:
-    with open(path, 'r') as f:
-      for line in f.readlines():
-        result = re.search(r"\[(.*?)\]\((.*?)\)", line)
-        if result and "ioit://" not in result.group(2):
-          uris.add(result.group(2))
-  return uris
+    """
+    Reads the collection of files passed as parameter and returns the set of uris found inside all the files
+    :param files: an iterable of file paths
+    :return: a set contain all uri found
+    """
+    uri_set = set()
+    for path in files:
+        with open(path, 'r') as f:
+            content = f.read()
+            uris = extract_uris(content)
+            uri_set = uri_set.union(uris)
+    return uri_set
 
 
 def test_http_uri(uri):
-  """
-  Tests the uri passed as a parameter making an http get request.
-  If it causes an exception or an error code it appends the url to the collection of invalid_uris
-  :param uri: the uri to test
-  :return: the uri if it is problematic, None otherwise
-  """
-  try:
-    r = requests.get(uri, headers=HEADERS, timeout=MAX_TIMEOUT)
-    if r.status_code != requests.codes.ok:
-      invalid_uris.append(uri)
-      return None
-  except:
-    print("failed to connect:" + uri)
-    # invalid_uris.append(uri)
-    return uri
+    """
+    Tests the uri passed as a parameter making an http get request.
+    If it causes an exception or an error code it appends the url to the collection of invalid_uris
+    :param uri: the uri to test
+    :return: the uri if it is problematic, None otherwise
+    """
+    try:
+        r = requests.get(uri, headers=HEADERS, timeout=MAX_TIMEOUT)
```

Comment:
@Undermaken 
I'm at all sure that this code's working on checking the response code, maybe this could be a better check:
```python
        r = requests.get(uri, headers=HEADERS, timeout=MAX_TIMEOUT)
        if not r.ok:
            return uri
        else:
            return None
```
What do you think?

Anyway now we return the uri or none we can remove `invalid_uris.append(uri)`

Reply:
@CrisTofani both points are good!
fixed within https://github.com/pagopa/io-app/pull/1793/commits/4fde51c56d13814f1db3c867c6a7a5e3a06b26b2